<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingPannel</class>
 <widget class="QWidget" name="SettingPannel">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>550</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>700</width>
    <height>550</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>设置</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resource/imagine_res/icon/icon.qrc">
    <normaloff>:/icon/set-up-dot.png</normaloff>:/icon/set-up-dot.png</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(255, 255, 255);</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="2,8">
   <property name="spacing">
    <number>6</number>
   </property>
   <item>
    <widget class="QListWidget" name="setting_list">
     <property name="minimumSize">
      <size>
       <width>100</width>
       <height>0</height>
      </size>
     </property>
     <property name="styleSheet">
      <string notr="true">border: 2px,solid,rgb(0,0,0);
border-radius: 10px;
background-color: qconicalgradient(cx:1, cy:0.909, angle:0, stop:0 rgba(128, 160, 191, 255), stop:1 rgba(255, 255, 255, 255));
color: rgb(255, 255, 255);
font: 700 14pt &quot;HarmonyOS Sans SC&quot;;
selection-background-color: rgb(139, 158, 178);

</string>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
     <property name="showDropIndicator" stdset="0">
      <bool>false</bool>
     </property>
     <property name="alternatingRowColors">
      <bool>false</bool>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="iconSize">
      <size>
       <width>16</width>
       <height>16</height>
      </size>
     </property>
     <property name="resizeMode">
      <enum>QListView::Adjust</enum>
     </property>
     <property name="spacing">
      <number>6</number>
     </property>
     <property name="selectionRectVisible">
      <bool>false</bool>
     </property>
     <property name="currentRow">
      <number>0</number>
     </property>
     <item>
      <property name="text">
       <string>通用设置</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>运行设置</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>高级</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>其它</string>
      </property>
     </item>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QStackedWidget" name="setting_page_stack">
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="gen_setting_pag">
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <property name="spacing">
          <number>25</number>
         </property>
         <item>
          <layout class="QHBoxLayout" name="gen_lan_lay" stretch="3,7">
           <item>
            <widget class="QLabel" name="gen_lan_la">
             <property name="text">
              <string>语言</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="gen_lan_cb">
             <item>
              <property name="text">
               <string>中文</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Engilsh</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="gen_theme_lay" stretch="3,7">
           <item>
            <widget class="QLabel" name="gen_theme_la">
             <property name="text">
              <string>主题</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="gen_theme_cb"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="gen_font_fam_lay" stretch="3,7">
           <item>
            <widget class="QLabel" name="gen_font_fam_la">
             <property name="text">
              <string>字族</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QFontComboBox" name="gen_font_fam_cb"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="gen_font_size_lay" stretch="3,7">
           <item>
            <widget class="QLabel" name="gen_font_size_la">
             <property name="text">
              <string>字体大小</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="gen_font_size_sp">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="value">
              <number>12</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="gen_win_size_lay" stretch="3,7">
           <item>
            <widget class="QLabel" name="gen_win_size_la">
             <property name="text">
              <string>窗口默认大小</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="gen_win_size_cb_lay" stretch="1,4,1,4">
             <item>
              <widget class="QLabel" name="gen_win_size_w_la">
               <property name="text">
                <string>宽度</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="gen_win_size_w_sp">
               <property name="minimum">
                <number>100</number>
               </property>
               <property name="maximum">
                <number>3000</number>
               </property>
               <property name="singleStep">
                <number>100</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="gen_win_size_h_la">
               <property name="text">
                <string>高度</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="gen_win_size_h_sp">
               <property name="minimum">
                <number>100</number>
               </property>
               <property name="maximum">
                <number>3000</number>
               </property>
               <property name="singleStep">
                <number>100</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>206</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="run_setting_pag">
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <property name="spacing">
          <number>25</number>
         </property>
         <item>
          <layout class="QHBoxLayout" name="run_cmd_start_lay" stretch="3,7">
           <item>
            <widget class="QLabel" name="run_cmd_start_la">
             <property name="text">
              <string>指令起始微指令序号</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="run_cmd_start_sp">
             <property name="maximum">
              <number>500</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="run_cmd_interval_lay" stretch="3,7">
           <item>
            <widget class="QLabel" name="run_cmd_interval_la">
             <property name="text">
              <string>指令运行间隔时间(ms)</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_13">
             <item>
              <widget class="QSlider" name="run_cmd_interval_slider">
               <property name="minimum">
                <number>100</number>
               </property>
               <property name="maximum">
                <number>5000</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="run_cmd_interval_value_la">
               <property name="text">
                <string>TextLabel</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="run_real_fresh_lay" stretch="0,0,0">
           <property name="spacing">
            <number>0</number>
           </property>
           <item>
            <widget class="QCheckBox" name="run_real_fresh_cb">
             <property name="text">
              <string>电路图实时刷新</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="run_real_fresh_la">
             <property name="text">
              <string>( 注: 资源占用量较大 )</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>318</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="adv_setting_pag">
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="spacing">
          <number>25</number>
         </property>
         <property name="leftMargin">
          <number>9</number>
         </property>
         <property name="topMargin">
          <number>9</number>
         </property>
         <property name="rightMargin">
          <number>9</number>
         </property>
         <property name="bottomMargin">
          <number>9</number>
         </property>
         <item>
          <widget class="QScrollArea" name="scrollArea">
           <property name="styleSheet">
            <string notr="true">#scrollarea{
	border-width:0px;
}</string>
           </property>
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="scrollAreaWidgetContents">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>554</width>
              <height>473</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <property name="spacing">
              <number>25</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <layout class="QHBoxLayout" name="adv_ram_lay" stretch="3,7">
               <item>
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>RAM内存设置</string>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="2,8">
                 <item>
                  <widget class="QLabel" name="adv_ram_row_la">
                   <property name="text">
                    <string>行数</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="adv_ram_row_sp">
                   <property name="maximum">
                    <number>500</number>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="adv_rom_lay" stretch="3,7">
               <item>
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>ROM内存设置</string>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="2,8">
                 <item>
                  <widget class="QLabel" name="adv_rom_la">
                   <property name="text">
                    <string>行数</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="adv_rom_row_sp">
                   <property name="maximum">
                    <number>300</number>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="adv_bits_lay" stretch="3,7">
               <item>
                <widget class="QLabel" name="adv_bits_la">
                 <property name="text">
                  <string>模拟环境位数</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="adv_bits_sp">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="readOnly">
                  <bool>true</bool>
                 </property>
                 <property name="specialValueText">
                  <string/>
                 </property>
                 <property name="accelerated">
                  <bool>false</bool>
                 </property>
                 <property name="suffix">
                  <string>Bits</string>
                 </property>
                 <property name="minimum">
                  <number>8</number>
                 </property>
                 <property name="maximum">
                  <number>64</number>
                 </property>
                 <property name="singleStep">
                  <number>8</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QCheckBox" name="adv_designer_cb">
               <property name="text">
                <string>开启设计器</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="verticalSpacer_3">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="other_setting_pag">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="spacing">
          <number>6</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="version_info">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);
border-color: rgb(0, 0, 0);
border: 2px,solid,rgb(0,0,0);
border-radius: 10px;
padding: 10px;</string>
           </property>
           <property name="text">
            <string>&lt;!doctype html&gt;
&lt;html&gt;
&lt;head&gt;
&lt;meta charset='UTF-8'&gt;&lt;meta name='viewport' content='width=device-width initial-scale=1'&gt;
&lt;title&gt;&lt;/title&gt;&lt;/head&gt;
&lt;body&gt;&lt;h1&gt;Version 1.4.2.2&lt;/h1&gt;
&lt;h2&gt;本次更新记录&lt;/h2&gt;
&lt;ul&gt;
&lt;li&gt;错误修复&lt;/li&gt;
&lt;li&gt;更漂亮的界面设计&lt;/li&gt;
&lt;li&gt;增加了更多的自由选择设置，现在你可以定制自己的模拟环境了&lt;/li&gt;
&lt;li&gt;移除了Him&lt;/li&gt;

&lt;/ul&gt;
&lt;h2&gt;相关信息&lt;/h2&gt;
&lt;ul&gt;
&lt;li&gt;开源协议：Apache License Version 2.0, January 2004.Apache: &lt;a href='http://www.apache.org/licenses/'&gt;http://www.apache.org/licenses/&lt;/a&gt;&lt;/li&gt;
&lt;li&gt;软件开源地址 &lt;a href='https://github.com/HaleyEhrich/DvccSimulationEnvironment'&gt;https://github.com/HaleyEhrich/DvccSimulationEnvironment&lt;/a&gt;&lt;/li&gt;

&lt;/ul&gt;
&lt;h2&gt;About Me&lt;/h2&gt;
&lt;ul&gt;
&lt;li&gt;联系方式：&lt;a href='haley_ehrich@outlook.com'&gt;haley_ehrich@outlook.com&lt;/a&gt;&lt;/li&gt;
&lt;li&gt;个人主页: &lt;a href='https://haleyehrich.github.io'&gt;https://haleyehrich.github.io&lt;/a&gt;&lt;/li&gt;

&lt;/ul&gt;
&lt;p&gt;&amp;nbsp;&lt;/p&gt;
&lt;/body&gt;
&lt;/html&gt;</string>
           </property>
           <property name="openExternalLinks">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="set_default_set_btn">
             <property name="text">
              <string>恢复默认设置</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="check_update_btn">
             <property name="minimumSize">
              <size>
               <width>80</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>检查更新</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="save_setting">
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>保存设置</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="cancel_edit">
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>取消</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../resource/imagine_res/icon/icon.qrc"/>
 </resources>
 <connections/>
</ui>
